### PR TITLE
fastlane

### PR DESCRIPTION
**Pull request by odra**
# Fastlane samples

## Changes 
* fastlane sample files for ios and android
* fastlane docs

I did not add fastane samples for cordova because it feels a bit redundant as cordova commands and workflow are very straightforward.

### Locally

Copy the fastlane folder to your project's source folder and run the fastlane commands described in its readme docs.

### Jenkins

Copy both fastlane folder and Jenkisfile file into your project folder and trigger it in a digger-jenkins instance.

Note: currently our plugins (android sign and xcode) do not handle credentials very well from a pipeline perspective which is the reason I am not using the fastlane sign functions in those Jenkinsfile - this might be something to look in the future since we can raplace almost all build/sign logic with fastlane.

### CODE REVIEW

**Comment by laurafitzgerald** [comment]
@odra :eyes:

File: fastlane/android/fastlane/report.xml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (laurafitzgerald) [inline fastlane/android/fastlane/report.xml]: I think this file should be removed and added to .gitignore?

File: fastlane/ios/fastlane/report.xml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (laurafitzgerald) [inline fastlane/ios/fastlane/report.xml]: same here?

**Comment by laurafitzgerald** [comment]
@odra Testing this [here](https://rhmap.osm1.skunkhenry.com/?title=laura-fastlane-test#projects/z5jhxrvebpuseejvt2ux5vdg/apps/z5jhxru4avbrk27ovbdv4nhn/build) Perhaps I've made some error in the verification steps. The error i'm seeing is `/tmp/workspace/laura-fastlane-test-android-gradle_z5jhxru4avbrk27ovbdv4nhn@tmp/durable-39e4106f/script.sh: line 2: fastlane: command not found` but I don't see the file `script.sh` in the app code or in the fastlane code.

**Comment by odra** [comment]
@laurafitzgerald ah sorry we don't have neither ruby or fastlane in our android slave image, let me pr it

**Comment by laurafitzgerald** [comment]
@odra Android application built successfully using Jenkinsfile implementing fastlane. moving on to iOS.

**Comment by laurafitzgerald** [comment]
@odra iOS Build Verification done on osm4 using this [app](https://github.com/laurafitzgerald/helloworld-ios) The build was successful

**Comment by laurafitzgerald** [comment]
@odra could we include a docs section which informs the user that they should copy the Jenkinsfile and the relevant fastlane folder to their application and build using Jenkins?

**Pull request merged**